server/neptune/workflows/internal/deploy/revision/queue/queue.go

```diff
@@ -19,28 +25,39 @@ type Deploy struct {
 	queue *priority
 
 	// mutable: default is unlocked
-	lock LockStatus
+	lock               LockState
+	lockStatusCallback func(workflow.Context, *Deploy)
 }
 
-func NewQueue() *Deploy {
+func NewQueue(callback func(workflow.Context, *Deploy)) *Deploy {
 	return &Deploy{
-		queue: newPriorityQueue(),
+		queue:              newPriorityQueue(),
+		lockStatusCallback: callback,
 	}
 
 }
 
-func (q *Deploy) SetLockStatusForMergedTrigger(status LockStatus) {
-	q.lock = status
+func (q *Deploy) GetLockState() LockState {
+	return q.lock
+}
+
+func (q *Deploy) SetLockForMergedQueue(ctx workflow.Context, state LockState) {
```

Comment:
nit: `SetLockForQueue`?

Reply:
I'm going to keep this is as is because it only locks merged items and doesn't lock manual items.